gateway/handler.go

```diff
@@ -331,7 +331,7 @@ func (i *handler) isTrustlessRequest(contentPath ipath.Path, responseFormat stri
 	// Only allow "/{#1}/{#2}"-like paths.
 	trimmedPath := strings.Trim(contentPath.String(), "/")
 	pathComponents := strings.Split(trimmedPath, "/")
-	if len(pathComponents) != 2 {
+	if responseFormat != carResponseFormat && len(pathComponents) != 2 {
```

Comment:
Can we add a regression test for this? (both positive and negative, across trustless formats)
- `/ipfs/cid?format=raw`  → ok
- `/ipfs/cid/subpath?format=raw`  → expect error due to deserialized resp. being off.
- `/ipfs/cid?format=car`  → ok
- `/ipfs/cid/subpath?format=car`  → ok